apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/TimestampBound.cs

```diff
@@ -28,26 +28,26 @@ namespace Google.Cloud.Spanner.Data
     /// </summary>
     public sealed class TimestampBound : IEquatable<TimestampBound>
     {
-        private TimestampBound(TimestampBoundMode mode)
-        {
-            Mode = mode;
-        }
+        // Determines whether a read-only transaction will return the read timestamp.
+        private bool _returnReadTimestamp = false;
 
-        private TimestampBound(TimestampBoundMode mode, TimeSpan staleness)
+        private TimestampBound(TimestampBoundMode mode, TimeSpan staleness,
+                               DateTime timestamp, bool returnReadTimestamp = false)
         {
-            if (staleness.Ticks < 0)
+            if (staleness != TimeSpan.Zero && staleness.Ticks < 0)
             {
                 throw new ArgumentOutOfRangeException(nameof(staleness), "Staleness must not be negative");
             }
-            Mode = mode;
-            Staleness = staleness;
-        }
 
-        private TimestampBound(TimestampBoundMode mode, DateTime timestamp)
-        {
-            GaxPreconditions.CheckArgument(timestamp.Kind == DateTimeKind.Utc, nameof(timestamp), "Timestamps must be expressed in UTC");
+            if (timestamp != DateTime.MinValue)
```

Comment:
I think it would be simpler to have a min-value-in-utc private static variable:

```csharp
private static readonly s_utcMinValue = new DateTime(1, 1, 1, 0, 0, 0, DateTimeKind.Utc);
```

Then use that everywhere we're otherwise using `DateTime.MinValue`. That way the timestamp will *genuinely* always be in UTC, which is nicer IMO.